src/protagonist/API/Features/OriginStrategies/Requests/DeleteCustomerOriginStrategy.cs

```diff
@@ -0,0 +1,59 @@
+﻿using API.Features.OriginStrategies.Credentials;
+using DLCS.AWS.S3;
+using DLCS.AWS.S3.Models;
+using DLCS.Core;
+using DLCS.Model.Customers;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace API.Features.OriginStrategies.Requests;
+
+public class DeleteCustomerOriginStrategy : IRequest<ResultMessage<DeleteResult>>
+{
+    public int CustomerId { get; }
+    public string StrategyId { get; }
+
+    public DeleteCustomerOriginStrategy(int customerId, string strategyId)
+    {
+        CustomerId = customerId;
+        StrategyId = strategyId;
+    }
+    
+    public class DeleteCustomerOriginStrategyHandler : IRequestHandler<DeleteCustomerOriginStrategy, ResultMessage<DeleteResult>>
+    {
+        private readonly DlcsContext dbContext;
+        private readonly ILogger<DeleteCustomerOriginStrategyHandler> logger;
+        private readonly CredentialsExporter credentialsExporter;
+        
+        public DeleteCustomerOriginStrategyHandler(DlcsContext dbContext, ILogger<DeleteCustomerOriginStrategyHandler> logger, CredentialsExporter credentialsExporter)
+        {
+            this.dbContext = dbContext;
+            this.logger = logger;
+            this.credentialsExporter = credentialsExporter;
+        }
+
+        public async Task<ResultMessage<DeleteResult>> Handle(DeleteCustomerOriginStrategy request,
+            CancellationToken cancellationToken)
+        {
+            var strategy = await dbContext.CustomerOriginStrategies.SingleOrDefaultAsync(
```

Comment:
We use entityframework-extensions in API and it has a [`DeleteFromQuery()`](https://entityframework-extensions.net/delete-from-query) method that saves loading the entity in the first place.

```cs
int deletedRows = await dbContext.CustomerOriginStrategies
  .Where(s => s.Customer == request.CustomerId && s.Id == request.StrategyId)
  .DeleteFromQueryAsync(cancellationToken);

if (deletedRows == 0) // return 404

await credentialsExporter.TryDeleteCredentials(strategy);
// return success
```

Reply:
Would this be appropriate in this instance? In this request we need to the retrieve the path stored in the origin strategy's `credentials` field.

Reply:
Ah, of course. Yeah this doesn't fit